packages/client/tests/functional/_utils/setupQueryEngine.ts

```diff
@@ -0,0 +1,29 @@
+import { enginesVersion } from '@prisma/engines'
+import { download } from '@prisma/fetch-engine'
+import type { Platform } from '@prisma/get-platform'
+import { getNodeAPIName } from '@prisma/get-platform'
+import { ClientEngineType } from '@prisma/sdk'
+import fs from 'fs-extra'
+import path from 'path'
+
+/**
+ * Ensures the correct Query Engine (`node-api`/`binary`) is present. This is required as
+ * normally the downloading of the required engine is done in `getGenerators`. As the test
+ * clients bypass this we need to ensure the correct engine is present.
+ * @param clientEngineType
+ * @param platform
+ */
+export async function setupQueryEngine(clientEngineType: ClientEngineType, platform: Platform) {
+  const engineDownloadDir = path.join(__dirname, '..', '..', '..')
+  const queryEngineLibraryPath = path.join(engineDownloadDir, getNodeAPIName(platform, 'fs'))
+  const queryEngineBinaryPath = path.join(
+    engineDownloadDir,
+    `query-engine-${platform}${platform === 'windows' ? '.exe' : ''}`,
+  )
```

Comment:
Out of scope of this PR, but: I've seen this repeated in one form or the other in different places in the codebase. I think this should be extracted to a common function similar to `getNodeAPIName`.

Reply:
I'll leave a todo to be tackled via tech debt :+1: